mechanicalsoup/form.py

```diff
@@ -303,9 +305,23 @@ def choose_submit(self, submit):
             form.choose_submit('form_name_attr')
             browser.submit_selected()
         """
+        # Since choose_submit is destructive, it doesn't make sense to call
+        # this method twice unless no submit is specified.
+        if self._submit_chosen:
+            if submit is None:
+                return
+            else:
+                raise Exception('Submit already chosen. Cannot change submit!')
+
+        # All buttons NOT of type (button,reset) are valid submits
+        inps = [i for i in self.form.select('input[type="submit"], button')
+                if i.get('type', '') not in ('button', 'reset')]
+
+        # If no submit specified, choose the first one
+        if submit is None and inps:
+            submit = inps[0]
```

Comment:
That's better than the `or`-chain I saw in the first commit, but I think this should raise when `len(inps) > 1`.